### DISCUSSION TITLE
sources to targets: method of calculation

### DISCUSSION THREAD

**Discussion by ThomasCombettes** [valhalla/valhalla#4166] [Q&A] [answered]
In "sources_to_targets" mode:

If I send the same 5 sources and the same 5 targets, the API will return a 5x5 matrix with many identical values.

 Internally, does Valhalla calculate each cell one by one, or is there a slightly intelligent algorithm that factors in identical routes and returns a 5x5 matrix by computing each route only once?

**Comment by nilsnolde** [comment]
Sounds like https://github.com/valhalla/valhalla/issues/4153 maybe?

If not, can you please give both, request & response?

**Reply by nilsnolde** [reply, marked as answer]
Ah are you saying that 5 same sources and 5 same targets are producing basically only 2 unique values? Well, that'd make sense, no?

We're really only doing a 1:many or many:many Dijkstra (which one is depending on costing model & sources/targets size). The 1:many is definitely doing 5 Dijkstras here, but it'll find all destinations at the same spot. The many:many I'm fairly sure it'll actually send off 10 half-Dijkstras with that combo which might in fact be slower than 1:many for bike/ped(/motorscooter?).

Why is that even useful to do 5 same locations to 5 other same locations though?

**Reply by ThomasCombettes** [reply]
That's not even useful I took this example to make things clear.

I'm trying to understand how to use it!

Thanks a lot for answering me 👍.
